### PR TITLE
Fix bug in FinishTrajectory logic

### PR DESCRIPTION
**Pull request by sebastianklose**
This PR adds additional bookkeeping for trajectories that we scheduled for
finishing.

In Node::RunFinalOptimization(...), we were calling FinishTrajectory for
every active trajectory (state == ACTIVE). Since the state only gets updated
once the corresponding worker for the FinishTrajectory task is
scheduled, we were potentially calling FinishTrajectory twice for a
single trajectory id.

Reproducible on master e.g. with 
```
roslaunch cartographer_ros offline_backpack_2d.launch bag_filenames:=b2-2016-02-02-14-01-56.bag no_rviz:=true
```

### CODE REVIEW

**Comment by sebastianklose** [comment]
@MichaelGrupp FYI - this was introduced after switching to use `TrajectoryState`

File: cartographer_ros/cartographer_ros/node.cc [hunk 1]
```diff
@@ -666,7 +668,8 @@ void Node::RunFinalOptimization() {
   {
     for (const auto& entry : map_builder_bridge_.GetTrajectoryStates()) {
       const int trajectory_id = entry.first;
-      if (entry.second == TrajectoryState::ACTIVE) {
+      if (entry.second == TrajectoryState::ACTIVE &&
+          trajectories_scheduled_for_finish_.count(entry.first) == 0) {
```
Comment (cschuet) [inline cartographer_ros/cartographer_ros/node.cc]: move the check into FinishTrajectoryUnderLock() and add error message there talking about pending finish?

Comment (cschuet) [reply cartographer_ros/cartographer_ros/node.cc]: And nice catch, BTW!

**Comment by MichaelGrupp** [comment]
Oops, didn't notice that it behaves like that. Thanks.

**Comment by cschuet** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user cschuet. Merge queue now has a length of 1.

**Pull request merged**